src/v/kafka/server/handlers/fetch.cc

```diff
@@ -281,6 +282,74 @@ make_partition_response_error(model::partition_id p_id, error_code error) {
     };
 }
 
+uint32_t
```

Comment:
sizes in the kafka:: space are all signed to keep compatibility with the Java layer. I think this should be signed.